inst/include/cpp11/protect.hpp

```diff
@@ -313,9 +313,7 @@ static struct {
 
     PROTECT(obj);
 
-    if (TYPEOF(list_) != LISTSXP) {
```

Comment:
The only worry I have about this is the increased overhead to lookup the preserve list on each call to `insert()`

Wonder if we should make this static so it is only initialized once in each compilation unit?

Reply:
Alright, I can do that